base/dcp_common.go

```diff
@@ -97,6 +98,7 @@ func NewDCPCommon(callback sgbucket.FeedEventCallbackFunc, bucket Bucket, maxVbN
 		lastCheckpointTime:     make([]time.Time, maxVbNo),
 		backfill:               &newBackfillStatus,
 		feedID:                 feedID,
+		groupID:                groupID,
```

Comment:
I'd prefer to store the computed prefix `DCPCheckpointPrefixWithGroupID(c.groupID)` here, both to avoid the string concatenation cost every time we read/write checkpoints, but also to make it explicit what DCPCommon is using groupID for.